csharp/src/Drivers/Apache/Hive2/HiveServer2Connection.cs

```diff
@@ -46,6 +49,30 @@ internal TCLIService.Client Client
             get { return this.client ?? throw new InvalidOperationException("connection not open"); }
         }
 
+        protected string? VendorVersion
+        {
+            get
+            {
+                if (_vendorVersion == null && TryGetInfoType(TGetInfoType.CLI_DBMS_VER, out string? value))
+                {
+                    _vendorVersion = value;
+                }
+                return _vendorVersion;
+            }
+        }
+
+        protected string? VendorName
+        {
+            get
+            {
+                if (_vendorName == null && TryGetInfoType(TGetInfoType.CLI_DBMS_NAME, out string? value))
```

Comment:
If there's an error getting the info type, this will attempt to get it again every time the VendorName property is used. Is this the desired behavior, or should we track it (say the error isn't transient but perhaps the backend isn't reporting this field).

Reply:
Now throwing an exception when failure status is detected.